### PR TITLE
TRAN-134:  bulk update for shipment cost, vehicles without pagination, revert migration changes with blank fields

### PR DESCRIPTION
**Pull request by sniedzielski**
part of TICKET: https://sd-transit.atlassian.net/browse/TRAN-134

changes:
- bulk update PUT endpoint for shipment cost, 
- added GET endpoint for vehicles without pagination (necessary for displaying vehicles in AddCostForm), 
- revert migration changes with blank fields

### CODE REVIEW

File: transit/models/item.py [hunk 1]
```diff
@@ -40,11 +40,11 @@ class Meta:
 
 
 class ItemDetails(BaseModel):
-    batch_number = models.CharField(db_column='BatchNumber', max_length=255, null=False)
+    batch_number = models.CharField(db_column='BatchNumber', max_length=255, null=False, blank=False)
 
-    expiry_date = models.DateTimeField(db_column='ExpiryDate', null=True)
-    manufacturing_date = models.DateTimeField(db_column='ManufacturingDate', null=True)
-    received_date = models.DateTimeField(db_column='ReceivedDate', null=True)
+    expiry_date = models.DateTimeField(db_column='ExpiryDate', null=True, blank=True)
```
Comment (dborowiecki) [inline transit/models/item.py:45]: Note: We should later on change dates and decimal fields to null. For now I'll merge this to be sure that frontend will work properly.

Comment (sniedzielski) [reply transit/models/item.py:45]: yes. We need to change as we discussed.

File: transit/rest_api/forms/shipment/shipment_details.py [hunk 2]
```diff
@@ -119,6 +123,35 @@ class Meta:
         }
 
 
+class ShipmentAddCostsSerializerWrapper(serializers.ModelSerializer):
+    """
+        Utilizes ShipmentDetailsSerializer for purpose of bulk update of shipments costs.
+    """
+    _service = ShipmentAddCostsService()
+
+    id = serializers.IntegerField()
```
Comment (dborowiecki) [inline transit/rest_api/forms/shipment/shipment_details.py:132]: Is this shipment ID? If that's bulk upload then ID should be list, if serilizer is for single shipment then I'm not sure if that's required. Shipment should be referenced under `details=True` (endpoint like api/shipment_cost/pk/add_cost_to_shipment, where pk is responsible for defining instance).

File: transit/rest_api/forms/shipment/shipment_details.py [hunk 3]
```diff
@@ -119,6 +123,35 @@ class Meta:
...
+    def update(self, instance, validated_data):
+        raise NotImplementedError('update() not supported in ShipmentAddCostsSerializer,'
+                                  ' data should be alternated using add_cost_to_shipment')
+
+    def create(self, validated_data):
+        raise NotImplementedError('create() not supported in ShipmentAddCostsSerializer,'
+                                  ' data should be alternated using add_cost_to_shipment')
+
+    def add_cost_to_shipment(self):
+        shipment = dict(self.validated_data)
```
Comment (dborowiecki) [inline transit/rest_api/forms/shipment/shipment_details.py:143]: Why are you casting validated data to dict?

File: transit/services/shipment_add_costs_service.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    @transaction.atomic
+    def add_cost_to_shipment(self, shipment):
+        """
+            Save costs to given shipment.
+
+            :param shipment: shipment to which costs will be assigned.
+        """
+
+        ShipmentDetails.objects.filter(id=shipment['id']).update(
```
Comment (dborowiecki) [inline transit/services/shipment_add_costs_service.py]: ID is unique, there should be get, also it'll be better to pass shipment object as an argument. We shouldn't expect some specific payload with ['id'], please find examples from other services.

**Review by dborowiecki — CHANGES_REQUESTED** [review]
Please find my comments, instead of using id from payload we should reference shipment as serializer.instance

File: transit/rest_api/forms/shipment/shipment_details.py (hunk 2 above)
Comment (sniedzielski) [reply transit/rest_api/forms/shipment/shipment_details.py:132]: It is a serializer for many shipments hence here we have id

**Comment by sniedzielski** [comment]
@dborowiecki I applied changes.

File: transit/rest_api/forms/shipment/shipment_details.py (hunk 2 above)
Comment (dborowiecki) [reply transit/rest_api/forms/shipment/shipment_details.py:132]: We can keep it in that case, thanks

**Review by dborowiecki — APPROVED** [review]

**Pull request merged**